addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -150,11 +175,50 @@ class PurchaseRequisitionLine(models.Model):
     product_qty = fields.Float(string='Quantity', digits=dp.get_precision('Product Unit of Measure'))
     price_unit = fields.Float(string='Unit Price', digits=dp.get_precision('Product Price'))
     qty_ordered = fields.Float(compute='_compute_ordered_qty', string='Ordered Quantities')
-    requisition_id = fields.Many2one('purchase.requisition', string='Purchase Agreement', ondelete='cascade')
+    requisition_id = fields.Many2one('purchase.requisition', required=True, string='Purchase Agreement', ondelete='cascade')
     company_id = fields.Many2one('res.company', related='requisition_id.company_id', string='Company', store=True, readonly=True, default= lambda self: self.env['res.company']._company_default_get('purchase.requisition.line'))
     account_analytic_id = fields.Many2one('account.analytic.account', string='Analytic Account')
     schedule_date = fields.Date(string='Scheduled Date')
     move_dest_id = fields.Many2one('stock.move', 'Downstream Move')
+    supplier_info_ids = fields.One2many('product.supplierinfo', 'purchase_requisition_line_id')
+
+    @api.multi
+    def create(self,vals):
+        res = super(PurchaseRequisitionLine, self).create(vals)
+        if res.requisition_id.state not in ['draft', 'cancel', 'done'] and not res.requisition_id.is_quantity_copy:
+            supplier_infos = self.env['product.supplierinfo'].search([
+                ('product_id', '=', vals.get('product_id')),
+                ('name', '=', res.requisition_id.vendor_id.id),
+            ])
```

Comment:
Strange to me. Why don't directly search for the requisition id there? It won't work for 2 purchases requisitions that have the same vendor/same product and won't raise a warning.